docker-java/src/test/java/com/github/dockerjava/cmd/StatsCmdIT.java

```diff
@@ -37,23 +36,52 @@ public void testStatsStreaming() throws InterruptedException, IOException {
 
         dockerRule.getClient().startContainerCmd(container.getId()).exec();
 
-        StatsCallbackTest statsCallback = dockerRule.getClient().statsCmd(container.getId()).exec(
-                new StatsCallbackTest(countDownLatch));
+        Boolean gotStats;
+        try (StatsCallbackTest statsCallback = dockerRule.getClient().statsCmd(container.getId()).exec(
+                new StatsCallbackTest(countDownLatch))) {
+            countDownLatch.await(3, TimeUnit.SECONDS);
+            gotStats = statsCallback.gotStats();
+
+            LOG.info("Stop stats collection");
+        }
+
+        LOG.info("Stopping container");
+        dockerRule.getClient().stopContainerCmd(container.getId()).exec();
+        dockerRule.getClient().removeContainerCmd(container.getId()).exec();
+
+        LOG.info("Completed test");
+        assertTrue("Expected true", gotStats);
+    }
+
+    @Test
+    public void testStatsNoStreaming() throws InterruptedException, IOException {
+        CountDownLatch countDownLatch = new CountDownLatch(NUM_STATS);
+
+        String containerName = "generated_" + new SecureRandom().nextInt();
```

Comment:
Unless you need to assert the name, you don't need to set the name and it will be auto generated anyways.
Other tests will be cleaned up soon as well.

Reply:
Removed setting name explicitly.